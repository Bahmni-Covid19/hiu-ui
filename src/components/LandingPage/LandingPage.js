import React, {
  Fragment
} from "react";
import Button from "@material-ui/core/Button";
import Modal from "@material-ui/core/Modal";

const LandingPage = () => {
  const [open, setOpen] = useState(false);

  handleOpen = () => {
    setOpen(true);
  };

  handleClose = () => {
    setOpen(false);
  };

  return (<div>
    <Button type="button"
      variant="contained"
      onClick={handleOpen}
      color="primary" >
      Request Consent </Button> <Modal aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
        open={
          open
        }
        onClose={
          handleClose
        } >
      <div style={
        modalStyle
      }
        className={
          classes.paper
        } >
        Test text </div> </Modal ></div>
  );
};

export default LandingPage;