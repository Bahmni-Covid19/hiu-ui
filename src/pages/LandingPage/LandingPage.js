import React, { useState, useEffect } from 'react';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Snackbar from '@material-ui/core/Snackbar';
import * as PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import RequestAccess from '../RequestAccess/RequestAccessContainer';
import LoadConsentsContainer from '../../components/ConsentsListTable/LoadConsentsContainer';
import Header from '../../components/Header';

function getModalStyle() {
  const top = 50;
  const left = 50;

  return {
    top: `${top}%`,
    left: `${left}%`,
    transform: `translate(-${top}%, -${left}%)`,
  };
}

const useStyles = makeStyles((theme) => ({
  paper: {
    position: 'absolute',
    width: 'calc(100vw - 100px)',
    maxWidth: 1024,
    height: 'calc(100vh - 100px)',
    overflow: 'auto',
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),

    [theme.breakpoints.down('sm')]: {
      width: 'calc(100vw - 40px)',
      height: 'calc(100vh - 40px)',
    },
  },
  button: {
    marginTop: 30,
    marginBottom: 30,
  },
}));

const LandingPage = ({
  success,
  onCreateConsentResetState,
  onSearchResetState,
  loadConsents,
  loadConfigValueSets,
  loadCMConfigurations,
}) => {
  useEffect(() => {
    loadConfigValueSets();
    loadCMConfigurations();
  }, []);
  const classes = useStyles();
  const [modalStyle] = React.useState(getModalStyle);
  const [modalOpen, setModalOpen] = useState(false);
  const [snackBarOpen, setSnackBarOpen] = useState(false);

  const handleClose = () => {
    setModalOpen(false);
    onCreateConsentResetState();
    onSearchResetState();
  };

  if (success && modalOpen) {
    handleClose();
    loadConsents();
    setSnackBarOpen(true);
  }

  const handleSnackBarClose = () => {
    setSnackBarOpen(false);
  };

  const handleOpen = () => {
    setModalOpen(true);
  };

  return (
    <div>
      <Snackbar
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        open={snackBarOpen}
        autoHideDuration={4000}
        onClose={handleSnackBarClose}
        message="Consent requested successfully!"
      />
      <Header />
      <Button
        className={classes.button}
        type="button"
        variant="contained"
        onClick={handleOpen}
        color="primary"
      >
        New Consent Request
      </Button>
      <Modal open={modalOpen} onClose={handleClose}>
        <div style={modalStyle} className={classes.paper}>
          <RequestAccess />
        </div>
      </Modal>
      <LoadConsentsContainer />
    </div>
  );
};

LandingPage.propTypes = {
  success: PropTypes.bool,
  onCreateConsentResetState: PropTypes.func.isRequired,
  onSearchResetState: PropTypes.func.isRequired,
  loadConsents: PropTypes.func.isRequired,
  loadConfigValueSets: PropTypes.func.isRequired,
};

LandingPage.defaultProps = {
  success: false,
};

export default LandingPage;
