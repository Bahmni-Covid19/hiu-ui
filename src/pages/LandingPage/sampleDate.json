[
  {
    "name": "Roshni Iyer",
    "jatayuId": "roshni.iyer57@ncg",
    "age": "57",
    "gender": "Female",
    "requestStatus": "Request Sent",
    "consentGrantOn": "",
    "consentExpiryOn": ""
  },
  {
    "name": "Reshma M",
    "jatayuId": "rrm@ncg",
    "age": "34",
    "gender": "Female",
    "requestStatus": "Request Sent",
    "consentGrantOn": "",
    "consentExpiryOn": ""
  },
  {
    "name": "Ram Joshi",
    "jatayuId": "ramjoshi@ncg",
    "age": "45",
    "gender": "Male",
    "requestStatus": "Request Sent",
    "consentGrantOn": "",
    "consentExpiryOn": ""
  }
]
