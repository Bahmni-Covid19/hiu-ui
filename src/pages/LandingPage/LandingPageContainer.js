import { connect } from "react-redux";
import LandingPage from "./LandingPage";
import { onCreateConsentResetState } from "../../redux/actions/createConsentActions";
import { onSearchResetState } from "../../redux/actions/searchPatientIdActions";
import { loadConsents } from "../../redux/actions/loadConsentsActions";
import { fetchValueSetsRequest as loadConfigValueSets } from "../../redux/actions/initAppActions"

const mapStateToProps = state => ({
  success: state.createConsent.success
});

const mapDispatchToProps = {
  onCreateConsentResetState,
  onSearchResetState,
  loadConsents,
  loadConfigValueSets
};

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);
