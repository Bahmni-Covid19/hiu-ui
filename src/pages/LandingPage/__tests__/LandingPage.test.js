import React from "react";
import { shallow } from "enzyme";
import Button from "@material-ui/core/Button";
import LandingPage from "../LandingPage";
import { Modal, Snackbar, jssPreset } from "@material-ui/core";

describe("Lading Page", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LandingPage />);
  });

  it("should render properley", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("Request Access: Modal, should be closed by default", () => {
    expect(wrapper.find(Modal).props().open).toBe(false);
  });

  it("Success Message: Snackbar, should be closed by default", () => {
    expect(wrapper.find(Snackbar).props().open).toBe(false);
  });

  it("New Consent Request: Button, should open Request Access modal on click", () => {
    wrapper.find(Button).props().onClick();
    expect(wrapper.find(Modal).props().open).toBe(true);
  });

  describe("After successful creation of consent", () => {
    let onCreateConsentResetState, onSearchResetState, loadConsents, wrapper;

    beforeAll(() => {
      onCreateConsentResetState = jest.fn();
      onSearchResetState = jest.fn();
      loadConsents = jest.fn();
      wrapper = shallow(
        <LandingPage
          success={true}
          onCreateConsentResetState={onCreateConsentResetState}
          onSearchResetState={onSearchResetState}
          loadConsents={loadConsents}
        />
      );
      wrapper.find(Button).props().onClick(); //Open Modal
    });

    it("Success Message: Snackbar, should popup after successfully requesting a consent", () => {
      const successMessageSnackbar = wrapper.find(Snackbar);
      expect(successMessageSnackbar.props().open).toBe(true);
      expect(successMessageSnackbar.props().message).toBe(
        "Consent requested successfully!"
      );
    });

    it("should reload the consent list", () => {
      expect(loadConsents).toHaveBeenCalled();
    });

    it("should reset previously created consent creation state on close", () => {
      wrapper.find(Modal).props().onClose();
      expect(onCreateConsentResetState).toHaveBeenCalled();
    });
  
    it("should reset the previously searched patient state on close", () => {
      wrapper.find(Modal).props().onClose();
      expect(onSearchResetState).toHaveBeenCalled();
    });
  });
});
