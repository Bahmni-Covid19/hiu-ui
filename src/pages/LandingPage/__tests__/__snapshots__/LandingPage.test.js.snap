// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lading Page should render properley 1`] = `
"<div>
  <WithStyles(ForwardRef(Snackbar)) anchorOrigin={{...}} open={false} autoHideDuration={4000} onClose={[Function: handleSnackBarClose]} message=\\"Consent requested successfully!\\" />
  <Header />
  <WithStyles(ForwardRef(Button)) className=\\"makeStyles-button-2\\" type=\\"button\\" variant=\\"contained\\" onClick={[Function: handleOpen]} color=\\"primary\\">
    New Consent Request
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Modal) open={false} onClose={[Function: handleClose]}>
    <div style={{...}} className=\\"makeStyles-paper-1\\">
      <Connect(RequestAccess) />
    </div>
  </ForwardRef(Modal)>
  <Connect(ConsentsListTable) />
</div>"
`;

exports[`Landing Page should render properley 1`] = `
"<div>
  <WithStyles(ForwardRef(Snackbar)) anchorOrigin={{...}} open={false} autoHideDuration={4000} onClose={[Function: handleSnackBarClose]} message=\\"Consent requested successfully!\\" />
  <Header />
  <WithStyles(ForwardRef(Button)) className=\\"makeStyles-button-2\\" type=\\"button\\" variant=\\"contained\\" onClick={[Function: handleOpen]} color=\\"primary\\">
    New Consent Request
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Modal) open={false} onClose={[Function: handleClose]}>
    <div style={{...}} className=\\"makeStyles-paper-1\\">
      <Connect(RequestAccess) />
    </div>
  </ForwardRef(Modal)>
  <Connect(ConsentsListTable) />
</div>"
`;

exports[`Landing Page should render properly 1`] = `
"<div>
  <WithStyles(ForwardRef(Snackbar)) anchorOrigin={{...}} open={false} autoHideDuration={4000} onClose={[Function: handleSnackBarClose]} message=\\"Consent requested successfully!\\" />
  <Header />
  <WithStyles(ForwardRef(Button)) className=\\"makeStyles-button-2\\" type=\\"button\\" variant=\\"contained\\" onClick={[Function: handleOpen]} color=\\"primary\\">
    New Consent Request
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Modal) open={false} onClose={[Function: handleClose]}>
    <div style={{...}} className=\\"makeStyles-paper-1\\">
      <WithStyles(ForwardRef(IconButton)) aria-label=\\"close\\" onClick={[Function: handleClose]} className=\\"makeStyles-closeButton-3\\">
        <CancelIcon />
      </WithStyles(ForwardRef(IconButton))>
      <Connect(RequestAccess) />
    </div>
  </ForwardRef(Modal)>
  <Connect(ConsentsListTable) />
</div>"
`;
